src/pkg/cli/composeUp.go

```diff
@@ -189,20 +190,27 @@ func WaitAndTail(ctx context.Context, project *compose.Project, client client.Fa
 	ctx, cancelTail := context.WithCancelCause(ctx)
 	defer cancelTail(nil) // to cancel WaitServiceState and clean-up context
 
+	deploymentStatusCh := make(chan error, 2)
```

Comment:
This is last referenced about 50 lines later. Do you think we can factor out a function to set up the channel, the wait group, the listener goroutines, and maybe return a cleanup function that we can call at the end?